### PR TITLE
Add `WriteBuffer` to provided absbraction for delta policy

### PR DESCRIPTION
**Pull request by congqixia**
See also #27675 
/kind enhancement

### CODE REVIEW

**Comment by congqixia** [comment]
This pr is based on #27873

File: internal/datanode/writebuffer/delta_buffer.go [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+	"math"
+
+	"github.com/milvus-io/milvus-proto/go-api/v2/msgpb"
+	"github.com/milvus-io/milvus-proto/go-api/v2/schemapb"
+	"github.com/milvus-io/milvus/internal/storage"
+	"github.com/milvus-io/milvus/pkg/util/paramtable"
+	"github.com/milvus-io/milvus/pkg/util/typeutil"
+)
+
+type DeltaBuffer struct {
```
Comment (jiaoew1991) [inline internal/datanode/writebuffer/delta_buffer.go:13]: does it need a mutex？

Comment (congqixia) [reply internal/datanode/writebuffer/delta_buffer.go:13]: Delta buffer is protected inside `WriteBuffer`. There shall be no direct operation outside of this package

File: internal/datanode/writebuffer/sync_policy.go [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+type SyncPolicy func(buffers []*segmentBuffer, ts typeutil.Timestamp) []int64
+
+func SyncFullBuffer(buffers []*segmentBuffer, _ typeutil.Timestamp) []int64 {
+	return lo.FilterMap(buffers, func(buf *segmentBuffer, _ int) (int64, bool) {
+		return buf.segmentID, buf.IsFull()
+	})
+}
+
+func GetSyncStaleBufferPolicy(staleDuration time.Duration) SyncPolicy {
```
Comment (xiaofan-luan) [inline internal/datanode/writebuffer/sync_policy.go:22]: where did we handle global memory pressure policy?

**Pull request merged**